pilot/pkg/status/reporter.go

```diff
@@ -276,7 +277,7 @@ type distributionEvent struct {
 }
 
 func (r *Reporter) QueryLastNonce(conID string, distributionType xds.EventType) (noncePrefix string) {
-	key := conID + distributionType
+	key := strings.Join([]string{conID, distributionType}, "~")
```

Comment:
I would prefer to make a function to do this so we don't forget to do it/have inconsistent logic?

Also, I think `conID + "~" + distributionType` is more efficient

Reply:
@howardjohn generally, it's a best choice by using `+`, but there are some posts show that `strings.Builder` is the best efficient. However, I will create a function for the key generation and use the general way for it.

Reply:
Can you share those? afaik in this case the compiler optimizes it. If that's not the case please let me. know since we make that assumption in other parts of the code

Reply:
Yeah, please refer to one of them here:  [golang-how-to-efficiently-concatenate-strings](https://medium.com/@shantanubansal/golang-how-to-efficiently-concatenate-strings-f2e51564f8d)

Reply:
Those benchmarks are not testing the same thing we do here. When you do `a + b + c`, the compiler can optimize it in ways that concat in a loop cannot

Reply:
okay, thanks for the deeper investigation into the string concat. And I submitted this PR again, and added a generic method for adding delimiter to status report of config distribution. Could you please help to review it?